### DISCUSSION TITLE
July 2020: webmail, custom MDA and python framework work · poolp.org

### DISCUSSION THREAD

**Discussion by poolpOrg** [poolpOrg/poolp.org#122] [Articles]

**Comment by jdelic** [comment]
@poolpOrg 
> I also worked a bit on the go-opensmtpd package and made progress on the filter API, the goal being to allow writing a filter without worrying about the low-level protocol. Instead of reading the protocol, parsing it and determining what should be called or not, a filter can register callbacks for specific events and implement these callbacks.

I have [very recently](https://github.com/jdelic/opensmtpd-filters-go/commit/7c45fad0362f31f92b25e7fc4d0d16761fc018ed) worked on abstracting the API version on [opensmtpd-filters-go](https://github.com/jdelic/opensmtpd-filters-go/). Usage example can be found here: [opensmtpd-filter-greylistd/main.go](https://github.com/jdelic/opensmtpd-filter-greylistd/blob/master/main.go).

I am looking forward to seeing on how you're going to approach the problem :).
